test(datepicker): improve the coverage in datepicker component

YisusW opened the pull request:
- [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated tests.
 - [x] added/updated API documentation.
 - [x] added/updated demos.

YisusW committed:
Create main.yml

YisusW committed:
Update main.yml

add test-coverage to trigger this action when push some changes about test in test branch

YisusW committed:
refactor(schematicc): rename constant in ng add index

YisusW committed:
Update main.yml

YisusW committed:
Merge pull request #1 from YisusW/action-new-branch

Action new branch

YisusW committed:
test(datepicker):  date picker component onSlectionDone

YisusW committed:
Merge branch 'test-coverage' of https://github.com/YisusW/ngx-bootstrap into test-coverage

YisusW committed:
chore(action): remove action in test-coverage branch

YisusW committed:
test(datepicker): add unit test on method onUpdate

YisusW committed:
test(datepicker): add unit test to component

YisusW committed:
test(datepicker): add unit test to register on change

YisusW committed:
test(datepicker): add unit test to registerOnTouched function

YisusW committed:
test(datepicker): add unit test to branch in function

YisusW committed:
test(datepicker): add unit test to missed branch

add unit test when the activeDate get new Date value

YisusW committed:
ci: revert coverage in local and html report

YisusW committed:
test(datepicker): refactor tests and replace undefined by other values

YisusW committed:
test(datepicker): fix do not set functions to null

YisusW committed:
test(datepicker): fix module dateoicker module

daniloff200 commented:
Hi, @YisusW !
Thanks for your contribution! 
I do see a bunch of commits, which actually not related to this PR.
Could you squash them? 
I also see, that unit tests are failed, could you try to fix them, please?

YisusW committed:
test(datepicker): fix custom element web component in unit test

YisusW committed:
fix(test): datepicker component in ci were failed

YisusW committed:
test(fix): the onlyCurrentMonth input has a value

YisusW committed:
test(fix): add common module from angular

YisusW commented:
Hey @daniloff200,

i do not why, the ci report many errors, but i can not replicate in the local environment. i would like to know how to do it.

because i am doing changes to CI and hope it works because jest works okay in local.

YisusW committed:
test(fix): add module

daniloff200 commented:
@YisusW  I can't really say, what's going on right now, but, I've just used `npm run test` on my local and got the same errors, as on CI

YisusW committed:
Merge branch 'valor-software:development' into test-coverage

YisusW commented:
> @YisusW I can't really say, what's going on right now, but, I've just used `npm run test` on my local and got the same errors, as on CI

after merge the origin repo i see the errors i will fix it soon as possible

YisusW committed:
test(datepicker): fix unit tests and add should create

the unit test just run with some changes in source code

YisusW committed:
refactor(datepicker): change binding on onlyCurrentMonth

add format in the component by prettier on vscode pluggin and fix unit test by the refactor

SvetlanaMuravlova committed:
Merge branch 'development' into test-coverage

YisusW committed:
Merge branch 'valor-software:development' into test-coverage

YisusW committed:
refactor(datepicker): change to old structure to see the real change

YisusW committed:
refactor(datepicker): revert some previous version changes

SvetlanaMuravlova commented on src/datepicker/datepicker.component.ts:
```diff
@@ -41,7 +41,7 @@ export const DATEPICKER_CONTROL_VALUE_ACCESSOR: Provider = {
                       [dateDisabled]="dateDisabled"
                       [dayDisabled]="dayDisabled"
-                      [onlyCurrentMonth]="onlyCurrentMonth"
+                      (onlyCurrentMonth)="onlyCurrentMonth"
```
@YisusW i think it isn't really correct improve
